File: AtCoder/AtCodersBignnersSelection/cpp/abc086c.cpp
#include <bits/stdc++.h>
using namespace std;

bool plan(int t, int x, int y) {
    t = abs(t);
    x = abs(x);
    y = abs(y);
    if ((x + y <= t) && ((x + y - t) % 2 == 0)) {
        return true;
    }
    else {
        return false;
    }
}


int main() {
    int N;
    cin >> N;
    vector<int> t(N + 1);
    vector<int> x(N + 1);
    vector<int> y(N + 1);
    for (int i = 0; i < N; i++){
        int ti, xi, yi;
        cin >> ti >> xi >> yi;
        t.at(i + 1) = ti - t.at(i);
        x.at(i + 1) = xi - x.at(i);
        y.at(i + 1) = yi - y.at(i);
    }
    bool flag = true;
    for (int i = 0; i < N; i++){
        if (not plan(t.at(i + 1), x.at(i + 1), y.at(i + 1))){
            flag = false;
            cout << "No" << endl;
            break;
        }
    }
    if (flag) {
        cout << "Yes" << endl;
    }
}

File: AtCoder/ABC/abc086/abc086c.py
N = int(input())
plan = [[0, 0, 0]] + [list(map(int, input().split())) for _ in range(N)]
diff = [[plan[i + 1][j] - plan[i][j] for j in range(3)] for i in range(N)]

def is_possible(t, x, y):
    if (t + x + y) % 2 == 0 and abs(x) + abs(y) <= t:
        return True
    else:
        return False

for i in range(N):
    ti, xi, yi = diff[i]
    if not is_possible(ti, xi, yi):
        print('No')
        exit()
print('Yes')